### PR TITLE
editor(440): add hyperlink capability

### LINKED ISSUE #440: EDITOR: Add hyperlink capability
#### Perceived Problem

- Currently, users cannot add hyperlinks to the Journaly Editor

#### Ideas / Proposed Solution(s)

- Let's add that as an option
- We should be able to follow existing patterns and reference the Slate docs & examples
- I'll need to add an icon, if someone picks this up in the meantime please tag me here or in the PR and I'll get you an icon 😄

### PR DESCRIPTION
**Pull request by davidsonsns**
## Description

**Issue:** closes #440 

It was made following slatejs link example: https://www.slatejs.org/examples/links

## Subtasks

- [x] I have added this PR to the Journaly Kanban project ✅
- [x] add svg link icon
- [x] add lib is-url
- [x] refactor some `nexus/utils/index.ts` functions to save hyperlinks
- [x] add link style on the editor
- [x] add link field type on `EditorNode`(`nexus/inputTypes.ts`)

## Screenshots
Screen Shot 2021-01-13 at 16 16 44

### CODE REVIEW

**Review by robin-macpherson — COMMENTED** [review]
Amazing @davidsonsns , I'm so excited for us to get this functionality into the writing experience! 

One question, automatically detecting a link and formatting it as such appears to be working perfectly 👌🏼 When I try to highlight some text and then click the hyperlink button - in this case I tried entering `www.google.com` - it formatted the text correctly but clicking the link doesn't seem to behave as expected. Here's a screen capture which you can also test in the staging environment:


https://user-images.githubusercontent.com/34203886/104526199-b8563b80-55b6-11eb-8ebf-a094280801a0.mov

**Comment by davidsonsns** [comment]
Interesting! I'll check it out.

**Comment by davidsonsns** [comment]
Just saw that the lib `xss` has a restriction related to the attribute `href`, https://github.com/leizongmin/js-xss/issues/118. To fix it, now I'm validating if the string has the protocol. If not, I'm adding `https://`.

I'd like to show a notification if the user adds an invalid URL. Is there something related to it? What do you think about this feedback.

**Comment by robin-macpherson** [comment]
@davidsonsns great! We have some patterns with some regexes and displaying errors when validating social media links that might be helpful! I’m currently out on a walk haha so it’s a bit tricky to link you directly right now but you can find them from the profile settings forms!

**Comment by robin-macpherson** [comment]
If you start from `SocialForm.tsx` you’ll be able to find them right away

**Review by Lanny — APPROVED** [review]
One small comment but not a blocker, good stuff!

I like the idea of giving some feedback around invalid urls. I personally thing good old `alert()` is perfectly acceptable although Robin might disagree. Maybe we could re-use toastify for error messaging?

File: components/JournalyEditor/JournalyEditor.tsx [hunk 1]
```diff
@@ -37,8 +40,10 @@ const HOTKEYS: { [key in HotKey]: string } = {
   'mod+u': 'underline',
 }
 
+type Type = 'mark' | 'block' | 'link'
```
[the remark this answers is not in the document]
Comment (davidsonsns) [reply components/JournalyEditor/JournalyEditor.tsx:43]: Lol. Agree 100%!

**Comment by davidsonsns** [comment]
@Lanny I don't like `alert()` as well LMAO 😄 . Just added `react-toastify` as used on `SocialForm.tsx` and it's working great, thanks @robin-macpherson for sending me it.

ready to review 💯

File: components/JournalyEditor/JournalyEditor.tsx [hunk 2]
```diff
@@ -58,7 +60,8 @@ const JournalyEditor: React.FC<JournalyEditorProps> = ({
 }: JournalyEditorProps) => {
   const renderElement = useCallback((props) => <Element {...props} />, [])
   const renderLeaf = useCallback((props) => <Leaf {...props} />, [])
-  const editor = useMemo(() => withHistory(withReact(createEditor())), [])
+  const editor = useMemo(() => withLinks(withHistory(withReact(createEditor()))), []) as Editor &
+    ReactEditor
```
Comment (robin-macpherson) [inline components/JournalyEditor/JournalyEditor.tsx:64]: Haha, this line is pretty amazing 😆

**Review by robin-macpherson — APPROVED** [review]
Amazing work @davidsonsns ! So glad to get this one through 🚀

Thanks for being so thorough with everything 😄

**Pull request merged**